src/connection/transport/sasl.rs

```diff
@@ -0,0 +1,25 @@
+#[derive(Debug, Clone)]
+pub enum SaslConfig {
+    Plain { username: String, password: String },
+}
+
+impl SaslConfig {
+    pub fn auth_bytes(&self) -> Vec<u8> {
```

Comment:
```suggestion
    pub(crate) fn auth_bytes(&self) -> Vec<u8> {
```

This is not a public API that falls under semantic versioning and that users should rely on. Same goes for `fn mechanism`.